hudi-common/src/main/java/org/apache/hudi/common/table/TableSchemaResolver.java

```diff
@@ -270,10 +280,11 @@ private MessageType getTableParquetSchemaFromDataFile() {
           if (instantAndCommitMetadata.isPresent()) {
             HoodieCommitMetadata commitMetadata = instantAndCommitMetadata.get().getRight();
             Iterator<String> filePaths = commitMetadata.getFileIdAndFullPaths(metaClient.getBasePathV2()).values().iterator();
-            return fetchSchemaFromFiles(filePaths);
+            return Option.of(fetchSchemaFromFiles(filePaths));
           } else {
-            throw new IllegalArgumentException("Could not find any data file written for commit, "
+            LOG.warn("Could not find any data file written for commit, "
```

Comment:
Why not throw exception? Does this behavior change impact functionality? I think the job would fail previously right.

Reply:
we are unifying the exception thrown. 
end user will still see schemaNotFoundError().